app/interactors/save_listing.rb

```diff
@@ -0,0 +1,15 @@
+class SaveListing < BaseInteractor
+  record :service_area
+  string :type
+
+  hash :person, strip: false
+
+  # todo: add other fields here and in nested interactors
+
+  def execute
+    merging_errors(in_transaction: true) do
+      person_record = compose SavePerson, person
```

Comment:
I was worried about the potential for weird nested transaction behavior here until I saw in the ActiveInteraction docs where `compose` will raise an error on failure like `.run!` 👍